tests/Doctrine/Tests/ORM/Persisters/BasicEntityPersisterCompositeTypeSqlTest.php

```diff
@@ -51,11 +52,9 @@ public function testSelectConditionStatementNeqNull()
         self::assertEquals('t0."admin1" IS NOT NULL AND t0."country" IS NOT NULL', $statement);
     }
 
-    /**
-     * @expectedException Doctrine\ORM\ORMException
-     */
     public function testSelectConditionStatementIn()
     {
+        $this->expectException(CantUseInOperatorOnCompositeKeys::class);
```

Comment:
Why converting it into a method here while keeping it in annotations on other places?

Reply:
Laziness/rabbit hole avoiding techniques I think

Reply:
Or was it maybe line length?

Reply:
The recommendation is to use the method instead of the annotation. It keeps expected results like exceptions and assertions in the test's code.

Reply:
Yeah and also, exception triggered *before* the call that expected to trigger them will not make the tests pass. But that's out of scope, sorry, that PR is big enough as is.